css/src/components/form/input.scss

```diff
@@ -0,0 +1,91 @@
+$input-color: $text !default;
+$input-background-color: $body-background !default;
+$input-border-color: $text !default;
+$input-placeholder-color: $text-subtle !default;
+
+$input-hover-color: $text !default;
+$input-hover-border-color: $primary !default;
+$input-focus-border-color: $primary !default;
+
+$input-disabled-color: $text-subtle !default;
+$input-disabled-background-color: $body-background-medium !default;
+$input-disabled-border-color: $table-border-dark !default;
+
+$input-error-border-color: $danger !default;
+$input-warning-border-color: $warning-active !default;
+$input-success-border-color: $score-high !default;
+
+.input {
+	@include control;
+
+	width: 100%;
+	max-width: 100%;
+	border-color: $input-border-color;
+	background-color: $input-background-color;
+	color: $input-color !important;
+
+	&::placeholder {
+		color: $input-placeholder-color;
+	}
+
+	&:hover,
+	&:focus-visible {
+		border-color: $input-hover-border-color;
+	}
+
+	&:focus-visible {
+		@extend %focus;
+
+		outline-color: $input-focus-border-color;
+		outline-offset: 0 !important;
+		outline-style: solid !important;
+	}
+
+	&[disabled] {
+		border-color: $input-disabled-border-color;
+		background-color: $input-disabled-background-color;
+		color: $input-disabled-color;
+	}
+
+	&.input-sm {
+		@include control-sm;
+	}
+
+	&.input-lg {
+		@include control-lg;
+	}
+
+	&.input-narrow {
+		width: auto;
+
+		&-tablet {
+			@include tablet {
+				width: auto;
+			}
+		}
+	}
+
+	&.input-error {
+		border-color: $input-error-border-color;
+
+		&:focus-visible {
```

Comment:
Need to determine the state of :focus-visible, is it implemented in all browsers in our matrix (Safari was last holdout that I know of). Remember that [we have logic](https://github.com/microsoft/atlas-design/blob/11f26a3e691344d61aff657e80b5d55182c49896/css/src/core/focus.scss#L1) to configure whether we use .focus-visible or :focus-visible. Docs uses the former.

Reply:
Consider as well whether you definitely require a separate class, or whether the rule can just be:

`&.&-error {
		border-color: $input-error-border-color;
        outline-color: $input-error-border-color;
}

Reply:
Partially updated. Still need to test the `:focus-visible` in different browsers.

Reply:
https://caniuse.com/?search=%3Afocus-visible

Looks like we're still missing Safari on iOS

Reply:
Here's an idea:

```
// /mixins/focus-visible.scss

@mixin focus-visible() {
	@if ($focus-visible-use-polyfill) {
		&.focus-visible {
			@content;
		}
	} @else {
		&:focus-visible {
			@content;
		}
	}
}

// elsewhere
.input {
    @include focus-visible {
	background-color: black !important;
    }
}